src/library.js

```diff
@@ -1566,30 +1566,34 @@ LibraryManager.library = {
 
   round__asm: true,
   round__sig: 'dd',
+  round__deps: ['llvm_copysign_f64'],
   round: function(d) {
     d = +d;
-    return d >= +0 ? +Math_floor(d + +0.5) : +Math_ceil(d - +0.5);
+    return +Math_trunc(d + +_llvm_copysign_f64(+0.49999999999999994, d));
```

Comment:
Note that this is handwritten JS code, and in particular, `llvm_copysign_*` is present here as a fallback, implementing it in JS when it can't be done otherwise. It's a slow code path, see https://github.com/kripken/emscripten/blob/incoming/src/library.js#L1559

To actually use copysign, this would need to be written in C, perhaps in `./system/lib/libc/extras.c`, and it could use a copysign intrinsic there.

Reply:
I think the CI failures are because this function is marked as asm.js, and I don't think `trunc` is a math intrinsic there.

Reply:
Yeah, between these two things and f32, it sounds like writing it in C is the better choice.

I could do it here with a ternary and floor, but it would be slightly inferior wasm, with a branch. I'll have a try at the C version.